### DISCUSSION TITLE
Make Fail

### DISCUSSION THREAD

**Discussion by 7W33D** [Arkq/bluez-alsa#639] [Q&A] [answered]
I'm trying to build BlueAlsa and it fails at:

/usr/bin/ld: a2dp-sbc.o: in function `a2dp_sbc_enc_thread':
.../bluez-alsa/build/src/../../src/a2dp-sbc.c:229: undefined reference to `sbc_reinit_a2dp'
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:685: bluealsa] Error 1

I built the latest SBC.
This is my configure:
../configure LDFLAGS="-L/usr/include/sbc

Feel like I'm probably making dumb mistake but any help
would be appreciated.
UbuntuStudio 20.04
automake (GNU automake) 1.16.1
Thanks!
David

**Comment by borine** [comment]
Your configure command is completely wrong. Please read the wiki page https://github.com/arkq/bluez-alsa/wiki/Installation-from-source

**Reply by AdoHal** [reply]
Hi, I followed all the instructions on the installation wiki page but I'm getting the same error. I've tried different configurations but nothing works and I always get this error. I'm new to raspberry so maybe I forgot to set something up. Any help or pointing in the right direction would be appreciated. Thanks

**Reply by paul-1** [reply, marked as answer]
This is actually and error in the sbc library.  The blues-alsa configure script doesn't require a new enough version. I just noticed this the other day, and have not got to submit a patch.

The SBC library needs to be > v1.5

Version 2.0 is available.  https://mirrors.edge.kernel.org/pub/linux/bluetooth/

This issue was patched after v1.4 https://git.kernel.org/pub/scm/bluetooth/sbc.git/commit/?id=8c31f87f40d4cfb2cba122ba89ada3062a2e4b49

**Reply by borine** [reply]
Ah, OK.

@7W33D is using Ubuntu 20.04 which has libsbc version 1.4, and that version results in the "undefined reference" error reported here. When you say you "built the latest SBC" which version did you actually build? You need libsbc version 1.5 or later for use with BlueALSA.

@paul-1 is correct, the BlueALSA configure.ac file needs to be updated to require libsbc version 1.5 or later: that requirement is because of commit 44a5dd9

**Reply by AdoHal** [reply]
Thanks, thats fixed the initial issue, but now I have a new one. I think it is for different question, but I can try to paste the output here. Maybe it is related. Has anyone encountered this ??

/usr/bin/ld: bluealsa-skeleton.o: in function `bluealsa_manager_iface_skeleton_new':
/home/pi/bluez-alsa/build/src/../../src/bluealsa-skeleton.c:48: undefined reference to `org_bluealsa_manager1_interface'
/usr/bin/ld: bluealsa-skeleton.o: in function `bluealsa_pcm_iface_skeleton_new':
/home/pi/bluez-alsa/build/src/../../src/bluealsa-skeleton.c:80: undefined reference to `org_bluealsa_pcm1_interface'
/usr/bin/ld: bluealsa-skeleton.o: in function `bluealsa_rfcomm_iface_skeleton_new':
/home/pi/bluez-alsa/build/src/../../src/bluealsa-skeleton.c:111: undefined reference to `org_bluealsa_rfcomm1_interface'
/usr/bin/ld: bluez-skeleton.o: in function `bluez_battery_provider_iface_skeleton_new':
/home/pi/bluez-alsa/build/src/../../src/bluez-skeleton.c:48: undefined reference to `org_bluez_battery_provider1_interface'
/usr/bin/ld: bluez-skeleton.o: in function `bluez_media_endpoint_iface_skeleton_new':
/home/pi/bluez-alsa/build/src/../../src/bluez-skeleton.c:79: undefined reference to `org_bluez_media_endpoint1_interface'
/usr/bin/ld: bluez-skeleton.o: in function `bluez_profile_iface_skeleton_new':
/home/pi/bluez-alsa/build/src/../../src/bluez-skeleton.c:110: undefined reference to `org_bluez_profile1_interface'

**Comment by paul-1** [comment]
Looks like you have old files in your build directory.  I would delete your build directory and then create a new one and build again.

**Reply by AdoHal** [reply]
I did it several times and it didnt help, but problem was propably with debian version. I was using Buster, so I installed newer version(Bullseye) and tried again. Everything works now, so I can play around. Once again, thanks.
